src/racing-car.js

```diff
@@ -26,43 +27,92 @@ class RacingCar {
     readline.question(
       '경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).\n',
       (names) => {
-        if (!this.validateInput(names)) {
-          this.exit();
+        if (!this.validateCarInput(names)) {
+          this.printWrongInput();
+          return this.start();
         }
 
-        this.printTitle();
+        readline.question('시도할 횟수는 몇회인가요?\n', (count) => {
+          if (!this.validateCountInput(count)) {
+            this.printWrongInput();
+            return this.start();
+          }
 
-        this.setRacers(names);
+          this.setCount(count);
 
-        this.race(this.racers);
-        this.printWinners();
+          this.printTitle();
 
-        readline.close();
+          this.setRacers(names);
+
+          this.race(this.racers);
+          this.printWinners();
+
+          readline.close();
+        });
       }
     );
   }
 
-  validateInput(names) {
-    const isValidated = names
+  validateCarInput(names) {
+    const isCarNameMinLengthValidated = names.length > 0;
+
+    if (!isCarNameMinLengthValidated) {
+      return false;
+    }
+
+    const isCarNameMaxLengthValidated = names
+      .split(',')
+      .every(
+        (name) => name.length > 0 && name.length <= this.#CAR_NAME_LENGTH_LIMIT
+      );
+
+    if (!isCarNameMaxLengthValidated) {
+      return false;
+    }
+
+    const isCarNameNotDuplicatedValidated =
+      new Set(names.split(',')).size === names.split(',').length;
+
+    if (!isCarNameNotDuplicatedValidated) {
+      return false;
+    }
+
+    const isCarNameNotIncludePunctualCharactersValidated = names
       .split(',')
-      .every((name) => name.length <= this.#CAR_NAME_LENGTH_LIMIT);
+      .every(
+        (name) =>
+          !name.match(/[\{\}\[\]\/?.,;:|\)*~`!^\-_+<>@\#$%&\\\=\(\'\"]/g)
```

Comment:
- escape을 안해도 되는 문자들이 보이는데 따로 의도하신 게 있는지 궁금합니다.
- global flag가 꼭 필요한 걸까요?
- regex도 따로 의미있는 변수에 담아 사용하시면 좋겠습니다.
```suggestion
          const invalidInputRegex = /[{}\[\]\/?.,;:|)*~`!^\-_+<>@#$%&\\=('"]/

          !name.match(invalidInputRegex)
```

Reply:
아무 생각 없이 구글링하면서 정규식을 통해 특수문자 제거하는 코드를 가져오다 보니 그렇게 되었네요..

Reply:
우선 당연히 글로벌 플래그는 필요 없습니다. 하나라도 있는지만 체크하면 되기 때문입니다.

Reply:
regex도 상수로 빼면 괜찮을 것 같아요. 특수문자이니

Reply:
불필요한 escape 문자 제거하였습니다.